### DISCUSSION TITLE
Can't spawn a fox

### DISCUSSION THREAD

**Discussion by Mmd4J** [Minestom/Minestom#1027] [Help] [answered]
Hello.
I've tried to spawn a fox but it didn't worked.
What's wrong with it ?
```Java
InstanceContainer instance = instanceManager.createInstanceContainer();
Entity entity = new Entity(EntityType.FOX);
entity.setInstance(instance, player.getPosition());
entity.spawn();
```

**Comment by ArvedEh** [comment, marked as answer]
The fox exists in a different instance (instances are like worlds in spigot)

Instead of creating a new instance for the fox, use the instance that the player is currently inside with player.getInstance() to spawn the fox in the same instance as the player.

**Comment by TheMode** [comment]
Should instead be
```java
InstanceContainer instance = player.getInstance();
Entity entity = new Entity(EntityType.FOX);
entity.setInstance(instance, player.getPosition());
```
`spawn` is already called internally, `setInstance` would be equivalent to `setWorld`.

**Comment by Mmd4J** [comment]
Thank you from both of you! 
sadly, i can only mark one of them as answer...
but yeah i was wrong in using them thanks
